### PR TITLE
License

### LINKED ISSUE #754: License
We already talked about choosing a license and we should do so. I suggest using either `GNU GPL v2.0` or the `MIT License`. As we all participated in this project, all of us need to agree to a license for Beagle to be under. Please proclaim your thoughts about either or both licenses here or suggest a different one.

### PR DESCRIPTION
**Pull request by jGleitz**
As I promised, this PR adds our License. It also fixes some errors in the build process that seem to have occurred because the software we’re using was updated.

I already removed the one file I knew not to comply to the EPL from the whole repo’s history.

This PR needs 5 LGTMs because it decides what may happen to your intellectual property, a decision you can only make for yourself.

Fixes #754

---

This change is [Reviewable](https://reviewable.io/reviews/beagle-pse/beagle/821)

### CODE REVIEW

**Comment by jGleitz** [comment]
Hey guys, I really can’t figure out how to make our project build again. I spend the whole night trying, and I failed. A lot of stuff is broken because software we use was updated. I was able to fix most parts but I’m stuck with wuff’s dependency resolving again.

I ’m removing the requirement for Travis to pass so we can merge this PR. The bottom line (at least for me) is that automatic builds for Eclipse are an unsolved task.

<details><summary>

<i>Previously, jGleitz (Joshua Gleitze) wrote…</i></summary>



> #### License
> 
> As I promised, this PR adds our License. It also fixes some errors in the build process that seem to have occurred because the software we’re using was updated.
> 
> I already removed the one file I knew not to comply to the EPL from the whole repo’s history.
> 
> This PR needs 5 LGTMs because it decides what may happen to your intellectual property, a decision you can only make for yourself.
> 
> Fixes #754
> 
> </details>

---

Review status: 0 of 13 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KI_HTExJGGybwMNzVOT:1422792685)_

**Comment by RomanLangrehr** [comment]
:lgtm: :white_check_mark: 

<details><summary>

<i>Previously, jGleitz (Joshua Gleitze) wrote…</i></summary>



> Hey guys, I really can’t figure out how to make our project build again. I spend the whole night trying, and I failed. A lot of stuff is broken because software we use was updated. I was able to fix most parts but I’m stuck with wuff’s dependency resolving again.
> 
> I ’m removing the requirement for Travis to pass so we can merge this PR. The bottom line (at least for me) is that automatic builds for Eclipse are an unsolved task.
> </details>

---

Review status: 0 of 13 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KIaslWu_hl7D05P_GTQ:-1603124213)_

**Comment by viridispirus** [comment]
:lgtm: 

<details><summary>

<i>Previously, RomanLangrehr (Roman Langrehr) wrote…</i></summary>



> :lgtm: :white_check_mark: 
> </details>

---

Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KIatS30EQ2JxKDDWrg4:980077423)_

**Comment by m1cm1c** [comment]
:lgtm:

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  b) a copy of this Agreement must be included with each copy of the Program.
+     Contributors may not remove or alter any copyright notices contained
+     within the Program.
+
+Each Contributor must identify itself as the originator of its Contribution,
+if
+any, in a manner that reasonably allows subsequent Recipients to identify the
+originator of the Contribution.
+
+4. COMMERCIAL DISTRIBUTION
```
Comment (flyfl) [inline LICENSE:101]: This is the only part of the license that is not working for me.
The chosen license is quite liberal (free usage, distribution, open source code etc..) and I think this suits Beagle. But it should enforce to handle further usage/distribution/licensing as liberal and free as we do. In concrete, I don't want our software to be commercial distributed in parts or as a whole.

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[LICENSE, line 101 [r1]](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-KIZhI7ZOwFgNDrjtZAs-r1-101:-KKc4laVmBGXjv16PKG_:-581035686) ([raw file](https://github.com/beagle-pse/beagle/blob/ee87bd4a88df6aef1d8815d8ab633344ea57e0ad/LICENSE#L101)):_

> In concrete, I don't want our software to be commercial distributed in parts or as a whole.

It’s your very right to not want this and we will have to consider another licence if you insist on that.

However, I’d like to ask you to reconsider your thinking. Because I guess the basic theme for you is: “If I do someting for free, why should somebody else make money out of it”? However, I think that it’s more complicated than that. For example, the Eclipse IDE was initially developed by IBM, major parts of the code base are still from (and often times maintained by) IBM employees. Nevertheless, Eclipse is free to use for anybody, even for commercial use. But IBM paid for it.

The industry is heavily investing into open source, and many of the free tools we use only exist because some company paid for their development (e.g. [Linux development](https://en.wikipedia.org/wiki/Linux_Foundation#Funding), [git development](http://sfconservancy.org/sponsors/)). Keeping that in mind, I think it’s a lot easier to accept that the industry is also _using_ open source: because a lot of them “give their share back”.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821)_

**Comment by flyfl** [comment]
:lgtm:

**Comment by jGleitz** [comment]
- @flyfl 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KLHCnMFGKrSKgkgq-KS:-978618705)_

**Comment by jGleitz** [comment]
-@flyfl 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KLHCsfZRVB5u-IeehxH:-1295140289)_

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[LICENSE, line 101 [r1]](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-KIZhI7ZOwFgNDrjtZAs-r1-101:-KLHDf-txSL0plrk8wTR:-499146009) ([raw file](https://github.com/beagle-pse/beagle/blob/ee87bd4a88df6aef1d8815d8ab633344ea57e0ad/LICENSE#L101)):_
<details><summary>

<i>Previously, jGleitz (Joshua Gleitze) wrote…</i></summary>



> > In concrete, I don't want our software to be commercial distributed in parts or as a whole.
> 
> It’s your very right to not want this and we will have to consider another licence if you insist on that.
> 
> However, I’d like to ask you to reconsider your thinking. Because I guess the basic theme for you is: “If I do someting for free, why should somebody else make money out of it”? However, I think that it’s more complicated than that. For example, the Eclipse IDE was initially developed by IBM, major parts of the code base are still from (and often times maintained by) IBM employees. Nevertheless, Eclipse is free to use for anybody, even for commercial use. But IBM paid for it.
> 
> The industry is heavily investing into open source, and many of the free tools we use only exist because some company paid for their development (e.g. [Linux development](https://en.wikipedia.org/wiki/Linux_Foundation#Funding), [git development](http://sfconservancy.org/sponsors/)). Keeping that in mind, I think it’s a lot easier to accept that the industry is also _using_ open source: because a lot of them “give their share back”.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821)_

**Comment by acBerger** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/beagle-pse/beagle/821#-:-KLY4jT0qcoZ3UfLXXJV:1417088721)_

**Pull request merged**